agent/snapshot.go

```diff
@@ -49,7 +49,28 @@ func (a *Agent) CancelBackup() {
 func (a *Agent) Backup(bcp pbm.BackupCmd) {
 	l := a.log.NewEvent(string(pbm.CmdBackup), bcp.Name)
 
-	q, err := backup.NodeSuits(a.node)
+	nodeInfo, err := a.node.GetInfo()
+	if err != nil {
+		l.Error("get node info: %v", err)
+		return
+	}
+
+	// In case there are some leftovers from the restore.
+	//
+	// There is no way to exclude some collections on mongodump stage
+	// while it accepts "exclude..." option only if sole db specified (¯\_(ツ)_/¯)
+	// So we're trying to clean up on primary node but because there is no guarantee
+	// that primary node reaches this code before some of the secondaries starts backup
+	// we have to double-check in backup.run() before mongodump and fail on dirty state.
+	if nodeInfo.IsPrimary && nodeInfo.Me == nodeInfo.Primary {
+		err = a.node.DropTMPcoll()
+		if err != nil {
+			l.Error("ensure no tmp collections: %v", err)
```

Comment:
This error message should list what the "temp" collections are. The user won't know.

Reply:
this error message contain collection names. in log it would be like:
```
backup: ensure no tmp collections: drop tmp roles collection pbmRRoles: blah-blah-blah
```

Reply:
And here is error would happen only if something went wrong during the collections removing (mongo error while Drop or so). We do double-check for the collection in the backup. And that error returns the collection name and asking to delete it manually if such collections left.

Reply:
Ah, so there would be proceeding log messages with the collection name(s).

Ah, sorry. I saw those "drop tmp users collection %s" messages during the first review but I thought they were all within the EnsureNoTMPcoll() function just due to my carelessness.